Include an example plugin for a key context

OdatNurd opened the pull request:
This is a simple plugin that was created in response to a request from
Troels on the Sublime Text Discord; it implements a custom key binding
context that allows you to detect if the selection(s) are reversed or
not.

OdatNurd committed:
Include an example plugin for a key context

This is a simple plugin that was created in response to a request from
Troels on the Sublime Text Discord; it implements a custom key binding
context that allows you to detect if the selection(s) are reversed or
not.

keith-hall commented on plugins/reversed_selection_listener.py:
```diff
@@ -0,0 +1,38 @@
+        if match_all:
+            rhs = all([r.a > r.b for r in view.sel()])
+        else:
+            rhs = view.sel()[0].a > view.sel()[0].b
```
maybe can re-use the expression above from line 34 but switch `any` for `all` - would that be more consistent with other ST keybindings when match_all is false?

keith-hall commented on plugins/reversed_selection_listener.py:
```diff
@@ -0,0 +1,38 @@
+        else:
+            rhs = view.sel()[0].a > view.sel()[0].b
+
+        return True if operator == sublime.OP_EQUAL and rhs == lhs else False
```
what if OP_NOT_EQUAL and rhs != lhs - that should also be true, right?

OdatNurd replied:
It turns out that this is more complicated than that, because if you're comparing for False (even when the operator is EQUALS), the state of the values in the array needs to be reversed.

Otherwise, the `any()` or `all()` tests for `True` but only finds `False`, and thinks that it should not trigger when in fact it should.

This one seems to work for all possible cases of the two operators and `match_all`.

I resolved the underlying issue by flipping the state of the operand when the operator is `!=` so that it always works out the same as an `==` operator will.

OdatNurd committed:
Rework conditional logic to work in all cases

This one was a bit of a brain burner, but the context will now properly
work whether match_all is True or False, and whether the actual thing
you're comparing for is True or False.

This is done by using any() or all() depending on match_all, and also
remembering that for both functions, they search for True.

Thus, the state of the booleans in the comparison list needs to be True
when the desired comparison direction is reversed, and also True when
the desired comparison direction is forward.

It's actually wether or not we check that final value against True or
False that matters, in the end.
